### PR TITLE
KT-14329  Do not report inspection "Remove empty class body" for anonymous objects

### PR DESCRIPTION
**Pull request by shiraji**
#KT-14329 Fixed

https://youtrack.jetbrains.com/issue/KT-14329

### CODE REVIEW

File: idea/src/org/jetbrains/kotlin/idea/intentions/RemoveEmptyClassBodyIntention.kt [hunk 1]
```diff
@@ -30,5 +32,10 @@ class RemoveEmptyClassBodyInspection : IntentionBasedInspection<KtClassBody>(Rem
 class RemoveEmptyClassBodyIntention : SelfTargetingOffsetIndependentIntention<KtClassBody>(KtClassBody::class.java, "Remove empty class body") {
     override fun applyTo(element: KtClassBody, editor: Editor?) = element.delete()
 
-    override fun isApplicableTo(element: KtClassBody) = element.text.replace("{", "").replace("}", "").isBlank()
+    override fun isApplicableTo(element: KtClassBody): Boolean {
+        element.getStrictParentOfType<KtObjectDeclaration>()?.let {
```
Comment (cypressious) [inline idea/src/org/jetbrains/kotlin/idea/intentions/RemoveEmptyClassBodyIntention.kt]: Won't
```
object Foo {
    class Bar {}
}
```
cause a false positive here?

Comment (shiraji) [inline idea/src/org/jetbrains/kotlin/idea/intentions/RemoveEmptyClassBodyIntention.kt]: Added test cases for following cases

```kotlin
// IS_APPLICABLE: false

object Foo {
    fun test() {
        object {}<caret>
    }
}

```

```kotlin
// IS_APPLICABLE: true

object Foo {
    class Bar {}<caret>
}
```

Comment (shiraji) [inline idea/src/org/jetbrains/kotlin/idea/intentions/RemoveEmptyClassBodyIntention.kt]: Ah!

```kotlin
object Foo {
    object {}<caret>
}
```

Comment (cypressious) [inline idea/src/org/jetbrains/kotlin/idea/intentions/RemoveEmptyClassBodyIntention.kt]: Is 
```kotlin
val foo = object {
    class Foo {}
}
```

valid?

**Comment by yole** [comment]
Thanks for your contribution!

**Pull request merged**